File: backend/corelibs/sistema.py
"""Funciones relacionadas con el sistema operativo."""

import os
import platform
import shlex
import subprocess


def obtener_os() -> str:
    """Retorna el nombre del sistema operativo."""
    return platform.system()


def ejecutar(comando: str) -> str:
    """Ejecuta un comando y devuelve su salida.

    Si el comando finaliza con un código de error se captura la excepción
    ``subprocess.CalledProcessError`` devolviendo ``stderr`` cuando esté
    disponible o lanzando un ``RuntimeError`` con información detallada.
    """
    args = shlex.split(comando)
    try:
        resultado = subprocess.run(
            args,
            check=True,
            text=True,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
        )
        return resultado.stdout
    except subprocess.CalledProcessError as exc:
        if exc.stderr:
            return exc.stderr
        raise RuntimeError(f"Error al ejecutar '{comando}': {exc}") from exc


def obtener_env(nombre: str) -> str | None:
    """Devuelve el valor de una variable de entorno."""
    return os.getenv(nombre)


def listar_dir(ruta: str) -> list[str]:
    """Lista los archivos de un directorio."""
    return os.listdir(ruta)

File: src/core/nativos/sistema.js
import os

import 'child_process'
import 'fs'
import 'os'
import child_process
import fs

export function obtener_os() {
    return os.platform();
}

export function ejecutar(args, permitidos = null) {
    if (permitidos && args.length && !permitidos.includes(args[0])) {
        throw new Error(`Comando no permitido: ${args[0]}`);
    }
    const [program, ...rest] = args;
    const proc = child_process.spawnSync(program, rest, { encoding: 'utf-8' });
    if (proc.error) throw proc.error;
    if (proc.status !== 0) throw new Error(proc.stderr);
    return proc.stdout;
}

export function obtener_env(nombre) {
    return process.env[nombre];
}

export function listar_dir(ruta) {
    return fs.readdirSync(ruta);
}
